File: problem/0766_ToeplitzMatrix.js
// 766_ToeplitzMatrix
// https://leetcode.com/problems/toeplitz-matrix/
// Given an m x n matrix, return true if the matrix is Toeplitz. Otherwise, return false.

// A matrix is Toeplitz if every diagonal from top-left to bottom-right has the same elements.

/**
 * @param {number[][]} matrix
 * @return {boolean}
 */
var isToeplitzMatrix = function(arr) {
  for(let i=0; i<arr.length-1; i++){ //檢查Y軸
    for(let j=0; j<arr[i].length-1;j++){ //檢查X軸
      if(arr[i][j] !== arr[i+1][j+1]) return false
    }
  }
  return true
};

// arr[0][0] -> arr[1][1] -> arr[2][2] 要一樣

console.log(isToeplitzMatrix([[1,2,3,4],[5,1,2,3],[9,5,1,2]])) //true
console.log(isToeplitzMatrix([[1,2],[2,2]])) //false

File: c#-solution/0766. Toeplitz Matrix.cs
// 766. Toeplitz Matrix
// https://leetcode.com/problems/toeplitz-matrix/description/
// Tag: C#, Array, Matrix, Easy

// 1. Brute Force
public class Solution {
    public bool IsToeplitzMatrix(int[][] matrix) {
        var m = matrix.Length;
        var n = matrix[0].Length;
        for(int i=0; i<m-1; i++){
            for(int j=0; j<n-1; j++){
                if(matrix[i][j] != matrix[i+1][j+1]) return false;
            }
        }
        return true;
    }
}